sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -52,14 +53,615 @@ def fix_big_data_passing(workflow: List[Dict[Text, Any]]) -> List[Dict[Text, Any
 
     Implementation:
 
-    1. Index the DAGs to understand how data is being passed and which inputs/outputs
+    1. Index the PIPELINEs to understand how data is being passed and which inputs/outputs
        are connected to each other.
     2. Search for direct data consumers in container/resource templates and some
-       DAG task attributes (e.g. conditions and loops) to find out which inputs
+       PIPELINE task attributes (e.g. conditions and loops) to find out which inputs
        are directly consumed as parameters/artifacts.
     3. Propagate the consumption information upstream to all inputs/outputs all
        the way up to the data producers.
-    4. Convert the inputs, outputs and arguments based on how they're consumed
-       downstream.
+    4. Convert the inputs, outputs based on how they're consumed downstream.
+    5. Use workspaces instead of result and params for bigger data passing.
+    6. Added workspaces to tasks, pipelines, pipelineruns, if the parmas is bigger data.
+    7. Convert "_" to "-" in the params and artifacts.
```

Comment:
Thanks @ckadner , Yes, I tested, we have to covert _ to - for the params, since the results value didn't support it, and we already coverted it, this will lead to inputs and outputs inconsistency. This was tested out for my big data passing cases, and also reproduced for watson cases.